nuxeo-distribution/nuxeo-server-hotreload-tests/src/test/java/org/nuxeo/ftest/server/hotreload/HotReloadTestRule.java

```diff
@@ -141,6 +141,29 @@ protected void deployDevBundle(Description description) {
         }
     }
 
+    /**
+     * Deploys the jar dev bundle located under src/test/resources/${TEST_CLASS_NAME}/{@code subPath} (if exists).
+     * <p />
+     * This method use {@link #updateDevBundles(String)}.
+     *
+     * @since 10.10
+     */
+    public void deployJarDevBundle(Class<?> testClass, String subPath) {
+        String className = testClass.getSimpleName();
+        String relativeBundlePath = "/" + className + "/" + subPath;
+        // #getResource could return null if resource doesn't exist
+        Optional<Path> bundlePathOpt = Optional.ofNullable(testClass.getResource(relativeBundlePath))
+                                               .map(URI_MAPPER)
+                                               .map(Paths::get)
+                                               .map(Path::toAbsolutePath)
+                                               .filter(p -> p.toFile().exists());
+        if (bundlePathOpt.isPresent()) {
```

Comment:
Given that you use `ifPresent` I'm not sure the complexity of the Optional is needed.

    URL url = testClass.getResource(relativeBundlePath);
    if (url != null) {
        Path path = Paths.get(URI_MAPPER.apply(url)).toAbsolutePath();
        updateDevBundles("Bundle:" + path);
    } else {
        throw new NuxeoException("Unable to locate JAR=" + subPath);
    }

But I see this idiom is already used by `deployDevBundle`...